### PR TITLE
fontra-pak (new cask)

### PR DESCRIPTION
**Pull request by Aaron-212**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version). (No Stable Version)
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

Also, Fontra Pak does have a GitHub Repository at <https://github.com/googlefonts/fontra-pak/>, but I think it is better to use <https://fontra.xyz/> as the homepage.

### CODE REVIEW

File: Casks/f/fontra-pak.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      verified: "fontra-download.black-foundry.com/"
+  name "Fontra Pak"
+  desc "Cross-platform bundled Fontra server for local desktop use"
+  homepage "https://fontra.xyz/"
+
+  depends_on macos: ">= :big_sur"
+
+  app "Fontra Pak.app"
+
+  zap trash: "~/Library/Saved Application State/xyz.fontra.fontra-pak.savedState"
```
Comment (razvanazamfirei) [inline Casks/f/fontra-pak.rb:15]:
```suggestion
  zap trash: [
    "~/Library/Preferences/com.xyz-fontra.FontraPak.plist",
    "~/Library/Preferences/xyz.fontra.fontra-pak.plist",
    "~/Library/Saved Application State/xyz.fontra.fontra-pak.savedState",
  ]
```

File: Casks/f/fontra-pak.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+cask "fontra-pak" do
+  version :latest
+  sha256 :no_check
+
+  url "https://fontra-download.black-foundry.com/FontraPak.dmg",
+      verified: "fontra-download.black-foundry.com/"
+  name "Fontra Pak"
+  desc "Cross-platform bundled Fontra server for local desktop use"
```
Comment (razvanazamfirei) [inline Casks/f/fontra-pak.rb:8]: Make sure the description matches our [guidelines](https://docs.brew.sh/Cask-Cookbook#stanza-desc).

**Review by razvanazamfirei — CHANGES_REQUESTED** [review]
Hi @Aaron-212, 

Thank you so much for opening this and making your first contribution. I made some suggestions.

My reservation for accepting it at this point is due to the lack of versioning. We use `:latest` when the version changes frequently, not when no version exists. 

I realize they're nightly builds; the app does have a dev version listen in the interface, but not in the Info.plist file. 

Would you be able to ask upstream to version their build more explicitly? Ideally, they would also provide a versioned url and a place where we could check the latest version.

Thank you again for your contributions on this!

**Comment by razvanazamfirei** [comment]
I guess, in theory, we could use some of the header information as a livecheck, but it doesn't seem like a great idea.
```ruby
  livecheck do
    url :url
    regex(/(\w*[._-]\w*)/i)
    strategy :header_match do |headers, regex|
      headers["etag"].scan(regex).map { |match| match[0].to_s }
    end
  end
```
Or trying to parse the `last-modified` header.

**Comment by krehel** [comment]
I'd also add, this application looks like it just installs a server that runs on localhost:8000 with a web editor?  Given there is source to build this, and isn't really a GUI application, this may be better off over in homebrew-core.

**Comment by Aaron-212** [comment]
> I'd also add, this application looks like it just installs a server that runs on localhost:8000 with a web editor?  Given there is source to build this, and isn't really a GUI application, this may be better off over in homebrew-core.

Maybe you're right, but the main repo for Fontra is also unversioned, can it still be accepted as a formula?

**Comment by p-linnane** [comment]
We can't accept software that isn't versioned in a meaningful way in any of our repos. Apologies, but passing on this.

**Pull request closed without merging**